x-pack/solutions/observability/plugins/observability/server/services/linked_dashboards_client.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import type { IContentClient } from '@kbn/content-management-plugin/server/types';
+import type { Logger, SavedObjectsFindResult } from '@kbn/core/server';
+import type { DashboardAttributes } from '@kbn/dashboard-plugin/server';
+import type { LinkedDashboard } from '@kbn/observability-schema';
+
+type Dashboard = SavedObjectsFindResult<DashboardAttributes>;
+
+export class LinkedDashboardsClient {
+  constructor(private logger: Logger, private dashboardClient: IContentClient<Dashboard>) {}
+
+  async getLinkedDashboardsByIds(ids: string[]): Promise<LinkedDashboard[]> {
+    const linkedDashboardsResponse = await Promise.all(
```

Comment:
How concerned should we be about the length of `ids` cause concurrent requests?

What I mean is, should we consider limits on the number of concurrent requests here via some simple batching?